src/scenes/Root/Header/HeaderSearch/HeaderSearch.tsx

```diff
@@ -25,14 +25,21 @@ const useStyles = makeStyles(({ palette, spacing }) => ({
 export const HeaderSearch: FC = () => {
   const classes = useStyles();
   const [{ q: search = '' }] = useSearch();
+  const { pathname, search: routeSearch } = useLocation();
+  const currentSearch = routeSearch.split('?q=')[1];
   const navigate = useNavigate();
 
   return (
     <Form
       initialValues={{ search }}
       onSubmit={({ search }) => {
         if (search) {
-          navigate(`/search?q=${search}`);
+          // cover cases where the user searches for the exact same thing again
+          if (pathname === '/search' && search === currentSearch) {
```

Comment:
I think `search` (from `useSearch`) will always equal `currentSearch`. The variable is just shadowed here in the submit handler. 

Would this work?
```tsx
onSubmit={({ search: newSearch }) => {
  if (newSearch === search) {
})
```